cmd/soroban-rpc/internal/events/events.go

```diff
@@ -207,16 +155,21 @@ func seek(events []event, cursor Cursor) []event {
 // IngestEvents adds new events from the given ledger into the store.
 // As a side effect, events which fall outside the retention window are
 // removed from the store.
-func (m *MemoryStore) IngestEvents(txReader *ingest.LedgerTransactionReader) error {
-	events, err := readEvents(txReader)
+func (m *MemoryStore) IngestEvents(networkPassphrase string, ledgerCloseMeta xdr.LedgerCloseMeta) error {
```

Comment:
The `networkPassphrase` here would never change with every ledgerClose event. We should be setting it up once.
Separately, the usage of that ( for cross referencing txHash ) is a bit.. questionable.
( i.e. it's not so obvious to me that the soroban-rpc need to perform validation on the incoming data )